### PR TITLE
chore: enable v1.23.15 & v1.24.9, use ubuntu 20.04 as default, force containerd runtime

### PR DESCRIPTION
**Pull request by christinalau0**
**Reason for Change**:

Add 1.23.15 and 1.24.9 on Azure Stack Hub
set ubuntu 20.04 as default on Azure Stack Hub
force overwrite from docker to containerd on Azure Stack Hub
fix rotate certs command

**Issue Fixed**:

The rotate certs command was failing on 20.04 environments, Output:
```
cp -p /etc/kubernetes/rotate-certs/certs/kubeconfig /root/.kube/config 
cp: cannot create regular file '/root/.kube/config': No such file or directory
```
The fix specifies the user instead of using ~, works on both 20.04 and 18.04

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [x] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [x] adds unit tests
- [x] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/api/common/versions.go [hunk 1]
```diff
@@ -389,12 +389,12 @@ var AllKubernetesSupportedVersionsAzureStack = map[string]bool{
 	"1.22.7":  false,
 	"1.22.15": false,
 	"1.22.16": false,
-	"1.22.17": true,
 	"1.23.6":  false,
 	"1.23.12": false,
 	"1.23.13": false,
 	"1.23.14": false,
 	"1.23.15": true,
+	"1.24.7":  false,
```
Comment (jadarsie) [inline pkg/api/common/versions.go:397]: Let's keep the ASH tables in sync with `AllKubernetesSupportedVersions`

Comment (christinalau0) [reply pkg/api/common/versions.go:397]: Disabled1.22.17 in AllKubernetesSupportedVersions to keep in sync

Comment (christinalau0) [reply pkg/api/common/versions.go:397]: Disabled 1.20.15 and 1.21.14 as well

**Review by jadarsie — APPROVED** [review]

**Pull request merged**